### PR TITLE
chore: add some `tracing` to project loading

### PR DESCRIPTION
**Pull request by davidbarsky**
I wanted to see what's happening during project loading and if it could be parallelized. I'm thinking maybe, but it's not this PR :)

### CODE REVIEW

File: crates/rust-analyzer/src/global_state.rs [hunk 1]
```diff
@@ -361,6 +365,7 @@ impl GlobalState {
             (change, modified_rust_files, workspace_structure_change)
         };
 
+        let _p = span!(Level::INFO, "GlobalState::process_changes/apply_change");
```
Comment (lnicola) [inline crates/rust-analyzer/src/global_state.rs:368]: `.entered()`?

Comment (davidbarsky) [reply crates/rust-analyzer/src/global_state.rs:368]: ugh, someone should make a lint for this...

File: crates/rust-analyzer/src/main_loop.rs [hunk 2]
```diff
@@ -980,6 +986,7 @@ impl GlobalState {
 
     /// Handles an incoming notification.
     fn on_notification(&mut self, not: Notification) -> anyhow::Result<()> {
+        let _p = span!(Level::INFO, "GlobalState::on_notification", not.method = ?not.method);
```
Comment (lnicola) [inline crates/rust-analyzer/src/main_loop.rs:989]:
```suggestion
        let _p = span!(Level::INFO, "GlobalState::on_notification", not.method = ?not.method).entered();
```

File: crates/rust-analyzer/src/main_loop.rs [hunk 3]
```diff
@@ -867,6 +872,7 @@ impl GlobalState {
 
     /// Registers and handles a request. This should only be called once per incoming request.
     fn on_new_request(&mut self, request_received: Instant, req: Request) {
+        let _p = span!(Level::INFO, "GlobalState::on_new_request", req.method = ?req.method);
```
Comment (lnicola) [inline crates/rust-analyzer/src/main_loop.rs:875]: One more :grinning:.

Comment (lnicola) [reply crates/rust-analyzer/src/main_loop.rs:875]: Gentle nudge @davidbarsky

Comment (davidbarsky) [reply crates/rust-analyzer/src/main_loop.rs:875]: whoops, sorry! forgot. i'll fix it.

**Review by lnicola — APPROVED** [review]

**Pull request merged**